pkg/spec/spec.go

```diff
@@ -0,0 +1,81 @@
+package spec
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+
+	"github.com/grafana/tanka/pkg/spec/v1alpha1"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+// list of deprecated config keys and their alternatives
+// however, they still work and are aliased internally
+var deprecated = []depreciation{
+	{old: "namespace", new: "spec.namespace"},
+	{old: "server", new: "spec.apiServer"},
+	{old: "team", new: "metadata.labels.team"},
+}
+
+// Parse parses the json `data` into a `v1alpha1.Config` object.
+// `name` is the name of the environment
+func Parse(data []byte, name string) (*v1alpha1.Config, error) {
+	v := viper.New()
+	v.SetConfigType("json")
+	if err := v.ReadConfig(bytes.NewReader(data)); err != nil {
+		return nil, err
+	}
+	return parse(v, name)
+}
+
+// ParseDir parses the given environments `spec.json` into a `v1alpha1.Config`
+// object with the name set to the directories name
+func ParseDir(baseDir string) (*v1alpha1.Config, error) {
+	fi, err := os.Stat(baseDir)
+	if err != nil {
+		return nil, err
+	}
+	if !fi.IsDir() {
+		return nil, errors.New("baseDir is not an directory")
+	}
+
+	v := viper.New()
+	v.SetConfigName("spec")
+	v.AddConfigPath(fi.Name())
```

Comment:
This doesn't seem equivalent to what it was before. If the baseDir was `/a/b/c`, before we would've added `/a/b` but now we just add `c`. Am I missing something?

Reply:
Nice catch! Assumed `fi.Name()` was the full path